index.php

```diff
@@ -4,7 +4,7 @@
 
 $writer = new \App\Writer\UserWriter(
     new \App\Repository\UserRepository(new \App\Model\User()),
-    new \Illuminate\Events\Dispatcher()
+    new \App\Event\LaravelDispatcherAdapter(new \Illuminate\Events\Dispatcher())
```

Comment:
**Suggestion:** In the instantiation of the `UserWriter` class, you are directly creating new instances of `UserRepository` and `LaravelDispatcherAdapter`. This tightly couples your code to these specific implementations, reducing flexibility. Consider using dependency injection to make your code more flexible and testable. [best practice]
```suggestion
$userRepository = new \App\Repository\UserRepository(new \App\Model\User());
$eventDispatcher = new \App\Event\LaravelDispatcherAdapter(new \Illuminate\Events\Dispatcher());

$writer = new \App\Writer\UserWriter($userRepository, $eventDispatcher);
```